crates/outbound-networking/src/lib.rs

```diff
@@ -8,31 +8,6 @@ pub const ALLOWED_HOSTS_KEY: MetadataKey<Vec<String>> = MetadataKey::new("allowe
 pub const SERVICE_CHAINING_DOMAIN: &str = "spin.internal";
 pub const SERVICE_CHAINING_DOMAIN_SUFFIX: &str = ".spin.internal";
 
-/// Checks address against allowed hosts
-///
-/// Emits several warnings
-pub fn check_url(url: &str, scheme: &str, allowed_hosts: &AllowedHostsConfig) -> bool {
-    let Ok(url) = OutboundUrl::parse(url, scheme) else {
-        terminal::warn!(
-            "A component tried to make a request to an url that could not be parsed {url}.",
-        );
-        return false;
-    };
-    let is_allowed = allowed_hosts.allows(&url);
-
-    if !is_allowed {
-        terminal::warn!("A component tried to make a request to non-allowed url '{url}'.");
-        let (scheme, host, port) = (url.scheme, url.host, url.port);
-        let msg = if let Some(port) = port {
-            format!("`allowed_outbound_hosts = [\"{scheme}://{host}:{port}\"]`")
-        } else {
-            format!("`allowed_outbound_hosts = [\"{scheme}://{host}:$PORT\"]` (where $PORT is the correct port number)")
-        };
-        eprintln!("To allow requests, add {msg} to the manifest component section.");
```

Comment:
This language is merged with the outbound HTTP warning language in `disallowed_host_callback` below. This is probably a bit of a regression around the "where $PORT is the correct port number" stuff, but I feel like we could probably do better there anyway.